test/test_grid_file.py

```diff
@@ -145,12 +150,19 @@ def test_grid_in_default_opts(self):
 
         self.assertEqual(None, a.filename)
         self.assertEqual(None, a.name)
-        a.filename = "my_file"
+        if _IS_SYNC:
+            a.filename = "my_file"
+        else:
+            a.set("filename", "my_file")
```

Comment:
Most (all?) of these can be change back to the `a.filename = "my_file"` version now, right?

Reply:
Trying to use `a.filename = "my_file"` gives the following error:

```
AttributeError: property 'filename' of 'AsyncGridIn' object has no setter
```

This is because all those properties are `_a_grid_in_property` instances, which don't have an async setter since setting does file IO.

Reply:
We should make the setters work now. They will only raise an error on async when the file is closed.

Reply:
Ah I misunderstood--makes sense, will do.